### PR TITLE
Amortization calculation

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
*Replace this paragraph with a description of what this PR is changing or adding, and why.*

*List which issues are fixed by this PR. You must list at least one issue.*

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

File: src/utils/amortization.utils.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	rate: Ratio,
+	numberOfPayments: number
+) => {
+	if (!rate) {
+		rate = new Ratio(1, 1);
+	}
+
+	if (!numberOfPayments) {
+		numberOfPayments = 1;
+	}
```
Comment (BouweCeunen) [inline src/utils/amortization.utils.ts]: Can these two things ever happen? Shouldn't we always pass valid values to this method?

**Review by BouweCeunen — APPROVED** [review]

File: src/utils/amortization.utils.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import { Ratio } from "@credix/credix-client";
+import Big from "big.js";
+import { round } from "./format.utils";
+
+export const calculateMonthlyPayment = (
+	principal: number,
+	rate: Ratio,
+	numberOfPayments: number
+) => {
+	if (!rate) {
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts]: rate can't be undefined or null according to the function signature

File: src/utils/amortization.utils.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		rate = new Ratio(1, 1);
+	}
+
+	if (!numberOfPayments) {
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts]: same here

File: src/utils/amortization.utils.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import { DAYS_IN_REPAYMENT_PERIOD } from "@consts";
+import { Ratio } from "@credix/credix-client";
+import Big from "big.js";
+import { round } from "./format.utils";
+
+export const calculateMonthlyPayment = (
+	principal: number,
+	rate: Ratio,
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts]: We never use the term rate in the program or client. we should guard the domain lingo across projects so we should use interest or interestPercentage. I do think that interestRate is the more appropriate term for what we are doing and I have noted that we should move the lingo towards using interestRate eventually.

File: src/utils/amortization.utils.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import { DAYS_IN_REPAYMENT_PERIOD } from "@consts";
+import { Ratio } from "@credix/credix-client";
+import Big from "big.js";
+import { round } from "./format.utils";
+
+export const calculateMonthlyPayment = (
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts:6]: add the relevant formulas from this page as comments https://www.investopedia.com/terms/a/amortization.asp#:~:text=How%20to%20Calculate%20Amortization%20of%20Loans&text=You'll%20need%20to%20divide,your%20loan%20term%20by%2012 and have the code reflect them.

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**